### PR TITLE
Address history dumping related feedback.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

PR addresses feedback from https://github.com/mapbox/mapbox-navigation-ios/pull/2930.

### CODE REVIEW

File: Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift [hunk 1]
```diff
@@ -22,19 +22,23 @@ class Navigator {
     static var tilesURL: URL? = nil
     
     /**
-     Path to the file where history could be stored when `Navigator.dumpHistory(_:)` is called.
+     Path to the directory where history could be stored when `Navigator.writeHistory(completionHandler:)` is called.
```
Comment (MaximAlien) [inline Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift]: We actually need directory, not file in this case.

File: Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift [hunk 2]
```diff
@@ -22,19 +22,23 @@ class Navigator {
...
+     when writing finishes.
      
-     - parameter completion: A block object to be executed when history dumping ends.
+     - parameter completionHandler: A block object to be executed when history dumping ends.
      */
-    func dumpHistory(_ completion: @escaping (String?) -> Void) {
+    func writeHistory(completionHandler: @escaping (URL?) -> Void) {
         historyRecorder.dumpHistory { (path) in
-            completion(path)
+            if let path = path, let historyFileURL = URL(string: path) {
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift:37]: Does the original string contain a file: URL or a file path?

```suggestion
            if let path = path, let historyFileURL = URL(fileURLWithPath: path) {
```

**Review by 1ec5 — APPROVED** [review]

File: Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift (hunk 2 above)
Comment (MaximAlien) [reply Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift]: `fileURLWithPath` works fine as well.

File: Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift [hunk 3]
```diff
@@ -107,7 +111,7 @@ class Navigator {
                                                 config: NavigatorConfig(),
                                                 customConfig: customConfig)
         
-        historyRecorder = HistoryRecorderHandle.build(forHistoryFile: Navigator.historyFilePath?.absoluteString ?? "", config: configFactory)
+        historyRecorder = HistoryRecorderHandle.build(forHistoryFile: Navigator.historyDirectoryURL?.absoluteString ?? "", config: configFactory)
```
Comment (bamx23) [inline Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift:114]: @MaximAlien It should be `Navigator.historyDirectoryURL?.path`, not `Navigator.historyDirectoryURL?.absoluteString`. `url.absoluteString` returns a string in format `file:///...`

**Pull request merged**